tools/kagami/README.md

```diff
@@ -1,94 +1,134 @@
 # Kagami (Teacher and Exemplar and/or Looking glass)
 
-Generate and validate the automatically generated data files shipped with Iroha.
+Kagami is a tool used to generate and validate automatically generated data files that are shipped with Iroha.
 
-### Building
+## Build
 
-Use
+From anywhere in the repository, run:
 
 ```bash
 cargo build --bin kagami
 ```
 
-anywhere in this repository. This will place `kagami` inside the `target/debug/` directory (from the root of the repository).
+This will place `kagami` inside the `target/debug/` directory (from the root of the repository).
 
-### Usage
+## Usage
 
-```bash
-kagami 2.0.0-pre-rc.3
-Soramitsu Iroha2 team (https://github.com/orgs/soramitsu/teams/iroha2)
-Tool generating the cryptorgraphic key pairs,
+Run Kagami:
 
-USAGE:
+```
 kagami <SUBCOMMAND>
+```
 
-OPTIONS:
--h, --help       Print help information
--V, --version    Print version information
+### Subcommands
 
-SUBCOMMANDS:
-crypto     Generate cryptorgraphic key pairs
-docs       Generate a Markdown reference of configuration parameters
-genesis    Generate a default genesis block that is used in tests
-help       Print this message or the help of the given subcommand(s)
-schema     Generate schema used for code generation in Iroha SDKs
-```
+|        Command        |                             Description                              |
+| --------------------- | -------------------------------------------------------------------- |
+| [`crypto`](#crypto)   | Generate cryptographic key pairs                                     |
+| [`docs`](#docs)       | Generate a Markdown reference of configuration parameters            |
+| [`genesis`](#genesis) | Generate the default genesis block that is used in tests             |
+| [`schema`](#schema)   | Generate the schema used for code generation in Iroha SDKs           |
+| [`tokens`](#tokens)   | Generate a list of predefined permission tokens and their parameters |
+| `help`                | Print the help message for the tool or a subcommand                  |
 
-#### Key generation
+## `crypto`
 
-With a few examples.
+The `crypto` command generate cryptographic key pairs using the given algorithm, private key, and seed.
 
-```bash
-$ ./kagami crypto
-```
+|     Option      |                     Description                      |                   Default value                    |           Type            |
+| --------------- | ---------------------------------------------------- | -------------------------------------------------- | ------------------------- |
+| `--algorithm`   | The algorithm used to generate the key-pair          | `iroha_crypto::ED_25519`                           | `iroha_crypto::Algorithm` |
```

Comment:
```suggestion
| `--algorithm`   | The algorithm used to generate the key-pair          | `iroha_crypto::ED_25519`                           | String                               |
```
The input expected from the CLI user here is a string representation of the algorithm

Reply:
fixed

Reply:
Additional thoughts.
- It might be helpful to list all available values for `--algorithm`
- Some value must be supplied for `--private_key` or `--seed`. They have no default values
- Either `--private_key` or `--seed` can be specified
- Looking at the `Default value` and `Type` columns, a user might type `kagami crypto --json true`, which is an invalid input

Reply:
I've tried to address these concerns

<img width="635" alt="image" src="https://user-images.githubusercontent.com/17865025/169279686-1f1691fe-bdd2-4d9f-a37a-d7b0d8ca46d2.png">